homeassistant/components/motionmount/number.py

```diff
@@ -0,0 +1,81 @@
+"""Support for MotionMount numeric control."""
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MotionMountCoordinator
+from .entity import MotionMountEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Vogel's MotionMount from a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    unique_id = format_mac(coordinator.mm.mac.hex())
+
+    async_add_entities(
+        [
+            MotionMountExtension(coordinator, unique_id),
+            MotionMountTurn(coordinator, unique_id),
+        ]
+    )
+
+
+class MotionMountExtension(MotionMountEntity, NumberEntity):
+    """The target extension position of a MotionMount."""
+
+    _attr_native_max_value = 100
+    _attr_native_min_value = 0
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_translation_key = "motionmount_extension"
+
+    def __init__(self, coordinator: MotionMountCoordinator, unique_id: str) -> None:
+        """Initialize Extension number."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{unique_id}-extension"
+
+    @property
+    def native_value(self) -> float:
+        """Get native value."""
+        return float(self.coordinator.data["extension"])
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        self.async_write_ha_state()
+
```

Comment:
```suggestion
```

Reply:
It doesn't work if I remove line 50.
I've checked similar integrations, and I think they work because the use polling, so HA assumes state is updated. If I remove line 50 HA never picks up the new value. Are you sure it should work with it?

Reply:
Hi @joostlek, I've performed some more testing: When I enable polling the `native_value` property is indeed called and the interface updated even without the call to `self.async_write_ha_state`. When I disable polling `native_value` is never called, unless I perform the call to `self.async_write_ha_state`. So I truly believe that call is necessary. 
Hence, I've marked this PR is ready for review.

Reply:
Ah, well, apart from the fact that the 'ready for review' button is gone....

Reply:
Let me boot my IDE. I think this is the default implementation for `_handle_coordinator_update`

Reply:
The devil is in the details! I was thinking along the lines of 'There is some other mechanism notifying HA', so I disabled the call to `self.async_write_ha_state`, not the whole callback.
You're absolutely right, that with the whole callback removed it works as intended. Sorry for my stubbornness.....

Reply:
https://github.com/home-assistant/core/blob/d0efea3dbd00188363bfd6f3200d8aef8826ae4f/homeassistant/helpers/update_coordinator.py#L477-L480

Reply:
Like these number entities are a subclass of `MotionMountEntity`, which is a subclass of `CoordinatorEntity`, so this function call should be called if it isn't overwritten

Reply:
Yes, I've removed the whole callback now and pushed the changes.

Reply:
aah right. Also, there's another one at the other number entity

Reply:
'haastige spoed is zelden goed' :(
Removed and pushed.